src/main/java/edu/iis/mto/time/Order.java

```diff
@@ -34,7 +42,7 @@ public void submit() {
 
     public void confirm() {
         requireState(State.SUBMITTED);
-        long hoursElapsedAfterSubmittion = subbmitionDate.until(LocalDateTime.now(), ChronoUnit.HOURS);
+        long hoursElapsedAfterSubmittion = subbmitionDate.until(Objects.requireNonNullElseGet(testTime, LocalDateTime::now), ChronoUnit.HOURS);
```

Comment:
to nie jest dobre rozwiązanie wprowadza pan logikę wyliczanie daty "pod testy". Nie chodziło o to, aby do konstruktora przekazać "przesuniętą datę" (albo null - co oznacza brak przesunięcia). Zamówienie powinno wyliczać czas zgodnie z założeniami, ale być zależna od "zegara", który w testach można "zamockować"